tree/ntuplebrowse/v7/test/ntuple_browse.cxx

```diff
@@ -0,0 +1,520 @@
+// This hides the TBrowser.h file for this unit test and allows to define an initializable custom TBrowser class.
+#define ROOT_TBrowser
+
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <vector>
+
+#include <TDirectory.h>
+#include <TFile.h>
+
+#include <ROOT/RBrowseVisitor.hxx>
+#include <ROOT/RNTuple.hxx>
+#include <ROOT/RNTupleBrowser.hxx>
+#include <ROOT/RNTupleModel.hxx>
+
+#include "gtest/gtest.h"
+
+
+using RNTupleReader = ROOT::Experimental::RNTupleReader;
+using RNTupleBrowser = ROOT::Experimental::RNTupleBrowser;
+using RNTupleWriter = ROOT::Experimental::RNTupleWriter;
+using RNTupleModel = ROOT::Experimental::RNTupleModel;
+using RNonFolder = ROOT::Experimental::RNTupleFieldElement;
+using RFolder = ROOT::Experimental::RNTupleFieldElementFolder;
+
+/* This test cannot test the entire TBrwoser-support, it has no access to the TBrowser GUI.
+ * It only test certain functions, things it cannot test are:
+ * - The way the ntuple is displayed in TBrowser.
+ * - If histogram is displayed correctly.
+ * - It can count how often the TBrowser::Add() was called, but not what happens when it's called.
+ *
+ * To check if tests were succesful, it often uses the member variable RNTupleBrowser::fUnitTest
+ * defined in RNTupleBrowser.hxx.
+ */
+
+namespace {
+  /**
+   * An RAII wrapper around an open temporary file on disk. It cleans up the guarded file when the wrapper object
+   * goes out of scope.
+   */
+class FileRaii {
```

Comment:
Let's change to the way `FileRaii` is now used in the other unit tests: Every unit test uses its own file name. Once the file guard is defined in the test, use `GetPath()` to avoid repetition of the string.